urc_tf/src/world_frame.cpp

```diff
@@ -0,0 +1,96 @@
+#include "world_frame.hpp"
+
+namespace world_frame {
+WorldFrameBroadcaster::WorldFrameBroadcaster(const rclcpp::NodeOptions &options)
+    : Node("world_frame", options) {
+  tf_broadcaster_ = std::make_shared<tf2_ros::TransformBroadcaster>(this);
+
+  gps_subscriber = create_subscription<sensor_msgs::msg::NavSatFix>(
+      "/fix", rclcpp::SystemDefaultsQoS(),
+      [this](const sensor_msgs::msg::NavSatFix msg) { GPSCallback(msg); });
+
+  waypoint_subscriber = create_subscription<urc_msgs::msg::Waypoint>(
+      "/waypoint", rclcpp::SystemDefaultsQoS(),
+      [this](const urc_msgs::msg::Waypoint msg) { WaypointCallback(msg); });
+
+  set_base_subscriber = create_subscription<std_msgs::msg::Bool>(
+      "/set_base", rclcpp::SystemDefaultsQoS(),
+      [this](const std_msgs::msg::Bool msg) { SetBaseCallback(msg); });
+
+  imu_subscriber = create_subscription<sensor_msgs::msg::Imu>(
+      "/imu/data", rclcpp::SystemDefaultsQoS(),
+      [this](const sensor_msgs::msg::Imu msg) { IMUCallback(msg); });
+
+  baseStationLat = -1000;
+  baseStationLng = -1000;
+  baseStationAlt = -1000;
+}
+
+void WorldFrameBroadcaster::SetBaseCallback(const std_msgs::msg::Bool &msg) {
+  this->baseStationLat = currentLat;
+  this->baseStationLng = currentLng;
+  this->baseStationAlt = currentAlt;
+}
+
+void WorldFrameBroadcaster::IMUCallback(const sensor_msgs::msg::Imu &msg) {
+  this->imu_x = msg.orientation.x;
```

Comment:
There's no need to unpack all of the IMU message here. Just save the whole thing as a `sensor_msgs::msg::Imu` type and later set `t.transform.rotation = imu_msg.orientation`. This is much cleaner since you never use the individual values!